controllers/volumereplicationgroup_controller.go

```diff
@@ -199,38 +203,159 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
 
 	log.Info("Processing VolumeReplicationGroup ", volRepGroup.Spec, " in ns: ", req.NamespacedName)
 
-	pvcList, err := v.HandlePersistentVolumeClaims(ctx, volRepGroup)
+	// examine the DeletionTimestamp to determine whether the object is under deletion
+	if !volRepGroup.GetDeletionTimestamp().IsZero() {
+		if err := v.runAndRemoveFinalizer(ctx, volRepGroup, vrgFinalizerName); err != nil {
+			return ctrl.Result{Requeue: true}, nil
+		}
+
+		// stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	// Add finalizer if not present
+	if err := v.addFinalizer(ctx, volRepGroup, vrgFinalizerName); err != nil {
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
Same here; please log it or `return ctrl.Result{}, err`

Reply:
ACK.